dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScope.kt

```diff
@@ -29,6 +31,43 @@ interface FeatureScope {
         callback: (DatadogContext, EventBatchWriter) -> Unit
     )
 
+    /**
+     * Write data to the datastore.
+     *
+     * @param T datatype of the data to write to the datastore.
+     * @param dataStoreFileName name of the datastore file as there could be multiple such files per feature.
+     * @param featureName of the calling feature, to determine the path to the datastore file.
+     * @param serializer to use to serialize the data.
+     * @param data to write.
+     */
+    fun <T : Any> writeToDataStore(
+        dataStoreFileName: String,
```

Comment:
This is a good point and there's a question if we should replace the slashes in the path or log an exception. I'm in favor of the latter, since otherwise we're causing the user to believe that there's a key in the datastore that doesn't actually exist as they provided it

Reply:
I've removed the validation now that this is called "key"